osis-app/src/main/java/com/scality/osis/healthcheck/VaultHealthIndicator.java

```diff
@@ -0,0 +1,45 @@
+/**
+ *Copyright 2021 Scality, Inc.
+ *SPDX-License-Identifier: Apache License 2.0
+ */
+
+package com.scality.osis.healthcheck;
+
+import com.scality.osis.ScalityAppEnv;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.stereotype.Component;
+import java.net.HttpURLConnection;
+
+@Component("vault")
+public class VaultHealthIndicator implements HealthIndicator {
+    private static final Logger logger = LoggerFactory.getLogger(VaultHealthIndicator.class);
+
+    @Autowired
+    private ScalityAppEnv appEnv;
+
+    @Override
+    public Health health() {
+
+        try {
+            // check if Vault Admin Endpoint is reachable
+            HttpURLConnection connection = (HttpURLConnection) new java.net.URL(appEnv.getPlatformEndpoint()).openConnection();
+            connection.setConnectTimeout(appEnv.getVaultHealthCheckTimeout());
+            connection.connect();
+
+            // check if Vault S3 Interface Endpoint is reachable
+            connection = (HttpURLConnection) new java.net.URL(appEnv.getS3InterfaceEndpoint()).openConnection();
```

Comment:
What about IAM/ADMIN interface?

Reply:
Fixed URL to `/_/healthcheck`

Reply:
`getS3InterfaceEndpoint` is Vault 8500 port which is S3 routes. This endpoint is used for AssumeRoleBackbeat.

For IAM/ADMIN `getPlatformEndpoint` is used which is port 8600.

In this healthcheck, we are verifying connection to Vault Admin endpoint and then Vault S3 endpoint